app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -141,19 +144,29 @@ export const snapKeyringBuilder = (
         const learnMoreLink =
           'https://support.metamask.io/hc/en-us/articles/360015289452-How-to-add-accounts-in-your-wallet';
 
+        // If snap is preinstalled and does not request confirmation, skip the confirmation dialog
+        const skipConfirmation =
+          isSnapPreinstalled(snapId) && !displayConfirmation;
         // Since we use this in the finally, better to give it a default value if the controller call fails
-        let confirmationResult = false;
+        let confirmationResult = skipConfirmation;
```

Comment:
The comment does not really "match" anymore, maybe something like:

```ts
// If confirmation dialog are skipped, we consider the account creation to be confirmed  
```

WDYT?

Reply:
Good catch! [e448aa4](https://github.com/MetaMask/metamask-extension/pull/24876/commits/e448aa4d896df0b93b46f5fa140b752078f581e7)